modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/odata/filter/expression/SOSQLExpressionVisitorImpl.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.object.storage.salesforce.internal.odata.filter.expression;
+
+import com.liferay.list.type.entry.util.ListTypeEntryUtil;
+import com.liferay.object.constants.ObjectFieldConstants;
+import com.liferay.object.model.ObjectField;
+import com.liferay.object.service.ObjectFieldLocalService;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.odata.filter.expression.BinaryExpression;
+import com.liferay.portal.odata.filter.expression.Expression;
+import com.liferay.portal.odata.filter.expression.ExpressionVisitException;
+import com.liferay.portal.odata.filter.expression.ExpressionVisitor;
+import com.liferay.portal.odata.filter.expression.ListExpression;
+import com.liferay.portal.odata.filter.expression.LiteralExpression;
+import com.liferay.portal.odata.filter.expression.MemberExpression;
+import com.liferay.portal.odata.filter.expression.MethodExpression;
+import com.liferay.portal.odata.filter.expression.PrimitivePropertyExpression;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author Paulo Albuquerque
+ */
+public class SOSQLExpressionVisitorImpl implements ExpressionVisitor<Object> {
+
+	public SOSQLExpressionVisitorImpl(
+		long objectDefinitionId,
+		ObjectFieldLocalService objectFieldLocalService) {
+
+		_objectDefinitionId = objectDefinitionId;
+		_objectFieldLocalService = objectFieldLocalService;
+	}
+
+	@Override
+	public String visitBinaryExpressionOperation(
+			BinaryExpression.Operation operation, Object left, Object right)
+		throws ExpressionVisitException {
```

Comment:
It seems that this exception is never thrown on this method, am I right?

Reply:
Its required keep this exception because of the method signature defined by interface